Panic in etcd/raft on cluster restart.

Sometimes on cluster restart etcd/raft panics:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7c6e3d]

goroutine 1584 [running]:
panic(0xb5f0c0, 0xc420012150)
        /home/moroz/go/src/runtime/panic.go:500 +0x1a1
github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft.(*raft).appendEntry(0xc42357c000, 0xc420d0e320, 0x1, 0x1)
        /home/moroz/project/workspace/src/github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft/raft.go:413 +0xdd
github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft.(*raft).becomeLeader(0xc42357c000)
        /home/moroz/project/workspace/src/github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft/raft.go:499 +0x21a
github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft.(*raft).campaign(0xc42357c000)
        /home/moroz/project/workspace/src/github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft/raft.go:506 +0x5ea
github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft.(*raft).Step(0xc42357c000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/moroz/project/workspace/src/github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft/raft.go:540 +0x170
github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft.(*node).run(0xc420d0e0f0, 0xc42357c000)
        /home/moroz/project/workspace/src/github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft/node.go:310 +0x642
created by github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft.RestartNode
        /home/moroz/project/workspace/src/github.com/docker/swarmkit/vendor/github.com/coreos/etcd/raft/node.go:215 +0x312
```
It seems like it will be fixed on etcd side tomorrow, this issue is for tracking in swarmkit.
Issue in etcd: coreos/etcd#6013
ping @abronan

--- a/manager/state/raft/raft.go
+++ b/manager/state/raft/raft.go
@@ -401,7 +401,9 @@ func (n *Node) Run(ctx context.Context) error {
 			// restoring from the state, campaign to be the
 			// leader.
 			if !n.restored {
-				if len(n.cluster.Members()) <= 1 {
+				// Node ID should be in the progress list to Campaign
+				_, ok := n.Node.Status().Progress[n.Config.ID]
+				if len(n.cluster.Members()) <= 1 && ok {
 					if err := n.Campaign(n.Ctx); err != nil {
 						panic("raft: cannot campaign to be the leader on node restore")
 					}

